### ISSUE TITLE
[BUG] Avoid generating illegal surrogates for list query parameters

### ISSUE THREAD

**Issue by gluhar2006** [schemathesis/schemathesis#1370]
**Describe the bug**
There could be an error like this: `UnicodeEncodeError: 'utf-8' codec can't encode character '\udf06' in position 17: surrogates not allowed`.

**To Reproduce**
OpenApi schema to reproduce:

```yaml
      schema:
        type: array
        items:
          type: string
```
or
```yaml
      schema:
        type: array
        items:
          type: string
          pattern: "^[^\\x00]*$"
```

**Expected behavior**
Avoid illegal surrogates generation for the described case

**Environment (please complete the following information):**
(generally unimportant)
- OS - does not matter
- python - does not matter
- Schemathesis version 3.12.2
- Spec version 3.0.0

**Comment by Stranger6667** [comment]
Hi @gluhar2006 

Thank you for reporting and for a quick fix! :) Indeed, the current version allows such cases

**Comment by Stranger6667** [comment]
The fix is released in `3.12.3`
